clang/test/CXX/dcl.decl/dcl.meaning/dcl.fct/p6-cxx23.cpp

```diff
@@ -0,0 +1,4 @@
+// RUN: %clang_cc1 -std=c++23 -fsyntax-only -verify %s
+
+auto x0 = requires (this int) { true; }; // expected-error {{a requires clause cannot have an explicit object parameter}}
+auto x1 = requires (int, this int) { true; }; // expected-error {{a requires clause cannot have an explicit object parameter}}
```

Comment:
Can you add a test for `template <this auto> void f()` ?

Reply:
@cor3ntin sure, but non-type template parameter parsing doesn't use `ParseParameterDeclarationClause` -- it is done in [`ParseNonTypeTemplateParameter`](https://github.com/llvm/llvm-project/blob/main/clang/lib/Parse/ParseTemplate.cpp#L895)